libyul/backends/wasm/EVMToEwasmTranslator.cpp

```diff
@@ -841,7 +841,13 @@ function callvalue() -> z1, z2, z3, z4 {
 	z1, z2, z3, z4 := mload_internal(0:i32)
 }
 function calldataload(x1, x2, x3, x4) -> z1, z2, z3, z4 {
```

Comment:
I'd suggest to take this into its own PR and also fix codecopy, datacopy, extcodecopy, returndatacopy to truncate based on size.

Reply:
I agree

Reply:
I also agree, but at first I want to fix the polyfills here and extract those into a new PR later. It is more easy to test the changes here. Maybe I will also extract the semantic tests for ewasm to an other PR. But for now  I use this PR to ensure that the changes work.